### PR TITLE
@W-9142043 "self" functionality and base element refactoring

### PR DESCRIPTION
**Pull request by lizaiv77**
- add "self" element functionality
- fix validations to allow not providing element name
- refactor base classes for PO to have access to public methods like isPublic, waitFor etc.

matching PR in JS https://github.com/salesforce/utam-js/pull/159/files

### CODE REVIEW

File: utam-compiler/src/test/java/utam/compiler/grammar/UtamMethodAction_Tests.java [hunk 1]
```diff
@@ -100,10 +104,9 @@ public void testGetComposeActionInvalidUtilityStatement() {
     new UtamElement(ELEMENT_NAME, "clickable", getSelector()).testTraverse(context);
     UtamMethodAction action =
             new UtamMethodAction(null);
-    UtamError e = expectThrows(
+    expectThrows(
```
Comment (jimevanssfdc) [inline utam-compiler/src/test/java/utam/compiler/grammar/UtamMethodAction_Tests.java:107]: We don't want to assert the right error is being thrown?

File: utam-compiler/src/test/java/utam/compiler/grammar/UtamMethodAction_Tests.java [hunk 2]
```diff
@@ -282,4 +285,20 @@ public void testGetParameterTypesForCustomAction() {
...
         is(containsInAnyOrder(PrimitiveType.BOOLEAN, SELECTOR)));
   }
+
+  @Test
+  public void testWaitWithIncorrectElementThrows() {
+    TranslationContext context = TestUtilities.getTestTranslationContext();
+    context.setElement(new Document());
+    new UtamElement(ELEMENT_NAME, "clickable", getListSelector()).testTraverse(context);
+    // document element does not have waitFor API
+    expectThrows(
```
Comment (jimevanssfdc) [inline utam-compiler/src/test/java/utam/compiler/grammar/UtamMethodAction_Tests.java:295]: I think it's important when we're unit testing that a method throws an exception, that it's throwing the _right_ exception. Usually checking the message of the exception is good enough for this, but I know it's helped me track down several issues when refactoring.

File: utam-core/src/test/java/utam/core/framework/base/ContainerElementPageObjectTests.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  public void testGetContainer() {
+    ContainerElement containerElement = mock(ContainerElement.class);
+    ContainerElementPageObject pageObject = new ContainerElementPageObject(containerElement);
+    assertThat(pageObject.getContainerElement(), is(sameInstance(containerElement)));
+  }
+
+  @Test
+  public void testUnsupportedMethods() {
+    ContainerElementPageObject pageObject = new ContainerElementPageObject(mock(ContainerElement.class));
+    assertThrows(pageObject::load);
```
Comment (jimevanssfdc) [inline utam-core/src/test/java/utam/core/framework/base/ContainerElementPageObjectTests.java:36]: Can we please move these to the Hamcrest matcher style (`expectThrows`, followed by `assertThat`), for consistency?

**Review by jimevanssfdc — CHANGES_REQUESTED** [review]

**Comment by lizaiv77** [comment]
@jimevanssfdc addressed comments, pls have another round. thx

**Comment by lizaiv77** [comment]
back to draft - discovered gap with nullable

**Comment by lizaiv77** [comment]
@jimevanssfdc ready for review when you get a chance

**Review by jimevanssfdc — APPROVED** [review]

**Pull request merged**